onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.h

```diff
@@ -363,7 +363,33 @@ class ConvolutionHelperBase
 
         ML_CHECK_VALID_ARGUMENT(inputDimensions.size() > NonspatialDimensionCount, "Input dimensions must be >= 3");
 
-        ResolvingPadding(inputDimensions);
+        if (hasDynamicPads)
+        {
+            MLOperatorTensor padsTensor = info.GetConstantInputTensor(2);
+            const std::vector<uint32_t>& padsTensorDimensions = padsTensor.GetShape();
+            ML_CHECK_VALID_ARGUMENT(padsTensorDimensions.size() == 1, "Pads dimensions must equal 1");
+            const size_t dimCount = padsTensorDimensions[0];
+            ML_CHECK_VALID_ARGUMENT(dimCount == 2 * NchwSpatialDimensionCount, "Pads count must equal 4");
```

Comment:
I think it's more like `dimCount == 2 * (inputDimensions.size() - m_kernel.spatialDimensionCount)`. Otherwise this breaks Conv1D and Conv3D cases.

Reply:
The pads are for the spatial dimensions, not the others.  The two is to handle starts and ends.  Only 2D is supported in the case of dynamic pads.